assets/src/edit-story/elements/text/display.js

```diff
@@ -142,14 +128,18 @@ function TextDisplay({
   });
 
   if (backgroundTextMode === BACKGROUND_TEXT_MODE.HIGHLIGHT) {
+    const foregroundContent = getValidHTML(content);
```

Comment:
Hmm. Looking at this again. I think this really needs useMemo.

Reply:
Implemented a reusable solution with memoization in b6468b0d.